Source/SwiftLintFramework/Models/YamlParser.swift

```diff
@@ -13,14 +13,45 @@ public struct YamlParser {
     public static func parse(_ yaml: String,
                              env: [String: String] = ProcessInfo.processInfo.environment) throws -> [String: Any] {
         do {
-            return try Yams.load(yaml: yaml, .default,
-                                 .swiftlintContructor(env: env)) as? [String: Any] ?? [:]
+            let parser = try Parser(yaml: yaml, resolver: .default, constructor: .swiftlintContructor(env: env))
+            guard let node = try parser.singleRoot() else {
+                return [:]
+            }
+
+            try node.validateDuplicates(keyPath: nil)
+
+            return node.any as? [String: Any] ?? [:]
+        } catch let error as YamlParserError {
+            throw error
         } catch {
             throw YamlParserError.yamlParsing("\(error)")
         }
     }
 }
 
+private extension Node {
+    func validateDuplicates(keyPath: String?) throws {
+        var visitedKeys: Set<String> = []
+        for (key, value) in mapping ?? [:] {
+            guard let key = key.string else {
+                continue
+            }
+
+            let (inserted, _) = visitedKeys.insert(key)
+            if !inserted {
+                var message = "Duplicated keys found: '\(key)'."
```

Comment:
`Node` has location information as `marker` property.
So, it can generate error message with location and ` error:` keyword  that is recognized by Xcode:
```
                var message = "\(key.mark?.line ?? 0):\(key.mark?.column ?? 0): error: Duplicated keys found: '\(key.string!)'."
```
<img width="772" alt="2019-02-04 20 30 43" src="https://user-images.githubusercontent.com/33430/52205834-c32ccf80-28bb-11e9-9f10-db829d2a0da8.png">

Yams is created to generate error messages that Xcode will recognize by obeying the above manners.